src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -260,7 +260,7 @@ public override void VisitBinaryExpression(BinaryExpressionSyntax node)
         public override void VisitDeclarationExpression(DeclarationExpressionSyntax node)
         {
             var argumentSyntax = (ArgumentSyntax)node?.Parent;
-            var argumentListSyntax = (BaseArgumentListSyntax)argumentSyntax?.Parent;
+            var argumentListSyntax = argumentSyntax.FirstAncestorOrSelf<BaseArgumentListSyntax>();
```

Comment:
I think we shouldn't be creating Out Variables for such cases, they should be created only for the Out arguments.   

---
In reply to: [90172205](https://github.com/dotnet/roslyn/pull/15548#discussion_r90172205) [](ancestors = 90172205,90163033)